File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def solution(numbers):
    arr = set()
    
    
    answer=0
    
    for i in range(1, len(numbers)+1):
        for j in permutations(numbers, i):
            num = int(''.join(j))
            if num != 1 and num != 0:
                arr.add(num)
    
    for i in arr:
        sosu = True
        for j in range(2, int(i**(1/2))+1):
            if i%j==0:
                sosu = False
                break
                
        if(sosu):
            answer+=1
    
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <set>
#include <cmath>

using namespace std;

set<int> s;

bool isPrime(int x) {
    if (x < 2) return false;
    bool flag = true;
    for(int i=2; i*i<=x; i++) {
        if (x%i == 0)
            flag = false;
    }
    return flag;
}

void dfs(string numbers, string current, vector<bool>& visited) {
    if (!current.empty()) {
        s.insert(stoi(current));
    }

    for (int i = 0; i < numbers.size(); i++) {
        if (!visited[i]) {
            visited[i] = true;
            dfs(numbers, current + numbers[i], visited);
            visited[i] = false;
        }
    }
}

int solution(string numbers) {
    vector<bool> visited(numbers.size(), false);
    dfs(numbers, "", visited);
    int answer = 0;
    for (int num : s) {
        if (isPrime(num)) answer++;
    }
    return answer;
}